tests-legacy/resources/modules/ps_banner/ps_banner.php

```diff
@@ -0,0 +1,265 @@
+
+                        if (!move_uploaded_file($_FILES['BANNER_IMG_'.$lang['id_lang']]['tmp_name'], dirname(__FILE__).DIRECTORY_SEPARATOR.'img'.DIRECTORY_SEPARATOR.$file_name)) {
+                            return $this->displayError($this->trans('An error occurred while attempting to upload the file.', array(), 'Admin.Notifications.Error'));
+                        } else {
+                            if (Configuration::hasContext('BANNER_IMG', $lang['id_lang'], Shop::getContext())
+                                && Configuration::get('BANNER_IMG', $lang['id_lang']) != $file_name) {
+                                @unlink(dirname(__FILE__) . DIRECTORY_SEPARATOR . 'img' . DIRECTORY_SEPARATOR . Configuration::get('BANNER_IMG', $lang['id_lang']));
+                            }
+
+                            $values['BANNER_IMG'][$lang['id_lang']] = $file_name;
+                        }
+                    }
+
+                    $update_images_values = true;
+                }
+
+                $values['BANNER_LINK'][$lang['id_lang']] = Tools::getValue('BANNER_LINK_'.$lang['id_lang']);
+                $values['BANNER_DESC'][$lang['id_lang']] = Tools::getValue('BANNER_DESC_'.$lang['id_lang']);
+            }
+
+            if ($update_images_values) {
+                Configuration::updateValue('BANNER_IMG', $values['BANNER_IMG']);
+            }
+
+            Configuration::updateValue('BANNER_LINK', $values['BANNER_LINK']);
+            Configuration::updateValue('BANNER_DESC', $values['BANNER_DESC']);
+
+            $this->_clearCache($this->templateFile);
+
+            return $this->displayConfirmation($this->trans('The settings have been updated.', array(), 'Admin.Notifications.Success'));
+        }
+
+        return '';
+    }
+
+    public function getContent()
+    {
+        return $this->postProcess().$this->renderForm();
+    }
+
+    public function renderForm()
+    {
+        $fields_form = array(
+            'form' => array(
+                'legend' => array(
+                    'title' => $this->trans('Settings', array(), 'Admin.Global'),
+                    'icon' => 'icon-cogs'
+                ),
+                'input' => array(
+                    array(
+                        'type' => 'file_lang',
+                        'label' => $this->trans('Banner image', array(), 'Modules.Banner.Admin'),
+                        'name' => 'BANNER_IMG',
+                        'desc' => $this->trans('Upload an image for your top banner. The recommended dimensions are 1110 x 214px if you are using the default theme.', array(), 'Modules.Banner.Admin'),
+                        'lang' => true,
+                    ),
+                    array(
+                        'type' => 'text',
+                        'lang' => true,
+                        'label' => $this->trans('Banner Link', array(), 'Modules.Banner.Admin'),
```

Comment:
Please remove the capital letter: _Banner link_, thanks!

Reply:
These are tests, no need Wording check, sorry for the inconvenience